File: frontend/src/reporting/guardrails.ts
const prohibitedClaims: Array<{ pattern: RegExp; replacement: string }> = [
  { pattern: /\bmedimos emociones? reales?\b/gi, replacement: "estimamos respuesta cerebral predicha" },
  { pattern: /\bleemos la mente\b/gi, replacement: "analizamos patrones comparativos de procesamiento" },
  { pattern: /\blee la mente\b/gi, replacement: "analiza patrones comparativos de procesamiento" },
  { pattern: /\bgarantizamos? (compra|conversi[oó]n|recuerdo|engagement)\b/gi, replacement: "planteamos una hipotesis de mejora creativa" },
  { pattern: /\bpredice (compra|conversi[oó]n|conducta|emoci[oó]n)\b/gi, replacement: "estima patrones de respuesta probable" },
  { pattern: /\bmanipulaci[oó]n subconsciente\b/gi, replacement: "optimizacion editorial responsable" },
  { pattern: /\bneuromarketing que mide emociones\b/gi, replacement: "pretest neurocognitivo in silico" },
];

export function reviewReportClaims(text: string) {
  let safeText = text;
  const findings: Array<{ pattern: string; replacement: string; count: number }> = [];

  prohibitedClaims.forEach(({ pattern, replacement }) => {
    const matches = safeText.match(pattern);
    if (!matches?.length) return;
    findings.push({ pattern: pattern.source, replacement, count: matches.length });
    safeText = safeText.replace(pattern, replacement);
  });

  return {
    status: findings.length > 0 ? "rewritten" as const : "passed" as const,
    text: safeText,
    findings,
  };
}

File: backend/app/services/guardrails.py
from __future__ import annotations

from dataclasses import dataclass
import re


PROHIBITED_CLAIMS: dict[str, str] = {
    r"\bmedimos emociones? reales?\b": "estimamos respuesta cerebral predicha",
    r"\bleemos la mente\b": "analizamos patrones comparativos de procesamiento",
    r"\blee la mente\b": "analiza patrones comparativos de procesamiento",
    r"\bgarantizamos? (compra|conversion|conversión|recuerdo|engagement)\b": "planteamos una hipotesis de mejora creativa",
    r"\bpredice (compra|conversion|conversión|conducta|emocion|emoción)\b": "estima patrones de respuesta probable",
    r"\bmanipulacion subconsciente\b": "optimizacion editorial responsable",
    r"\bmanipulación subconsciente\b": "optimizacion editorial responsable",
    r"\bneuromarketing que mide emociones\b": "pretest neurocognitivo in silico",
}


@dataclass(frozen=True)
class GuardrailResult:
    status: str
    text: str
    findings: list[dict[str, str]]


def review_claims(text: str) -> GuardrailResult:
    findings: list[dict[str, str]] = []
    rewritten = text
    for pattern, replacement in PROHIBITED_CLAIMS.items():
        matches = list(re.finditer(pattern, rewritten, flags=re.IGNORECASE))
        if not matches:
            continue
        findings.append({"pattern": pattern, "replacement": replacement, "count": str(len(matches))})
        rewritten = re.sub(pattern, replacement, rewritten, flags=re.IGNORECASE)

    if findings:
        return GuardrailResult(status="rewritten", text=rewritten, findings=findings)

    return GuardrailResult(status="passed", text=text, findings=[])
